examples/scripts/project/categorization/bootstrap_taxonomy.py

```diff
@@ -0,0 +1,169 @@
+            f"A dataset with name {dataset_name} already exists. Try again after deleting the "
+            "dataset."
+        )
+        LOGGER.error(dataset_exists_error)
+        raise RuntimeError(dataset_exists_error)
+
+    # Create a dictionary of full path between string and list (used as label path)
+    taxonomy_dict = {", ".join(category): category for category in category_list}
+
+    # Create a dataframe
+    df = pd.DataFrame(list(taxonomy_dict.items()), columns=COLUMN_NAMES)
+
+    # Add category data and hash for tamr_id
+    df[CATEGORY_ATTRIBUTE_NAME] = df[FULL_PATH_LIST_NAME].apply(lambda x: x[-1])
+    df[PK_NAME] = df[FULL_PATH_STR_NAME].apply(lambda x: hash(x))
+    df.drop(FULL_PATH_LIST_NAME, axis=1, inplace=True)
+
+    # Create a dataset in Tamr
+    taxonomy_dataset = project.client.datasets.create_from_dataframe(
+        df, primary_key_name=PK_NAME, dataset_name=dataset_name
+    )
+    LOGGER.info(f"Created a dataset in Tamr with name: {taxonomy_dataset.name}")
+
+    # Add the dataset into the project
+    project.add_input_dataset(taxonomy_dataset)
+    LOGGER.info(f"Added {taxonomy_dataset.name} to project {project.name}")
+
+    # Create a unified attribute to map taxonomy categories
+    unified_attribute_name = "Taxonomy Categories"
+    assert schema.create_unified_attribute(project, unified_attribute_name=unified_attribute_name)
+
+    # Set ML configuration for the newly created unified attribute
+    assert schema.set_unified_attribute_configurations(
+        project,
+        unified_attribute_name=unified_attribute_name,
+        similarity_function="COSINE",
+        tokenizer="DEFAULT",
+        attribute_role="",
+        is_numeric=False,
+    )
+    LOGGER.info(f"Created a unified attribute {unified_attribute_name}")
+
+    # Tamr_id management is needed for label insert
+    all_tx = tbox.project.schema_mapping.transformations.get_all(project)
+    new_input_tx = tbox.project.categorization.transformations.InputTransformation(
+        "SELECT *, to_string(origin_entity_id) AS tamr_id;", [taxonomy_dataset]
+    )
+    all_tx.input_scope.insert(0, new_input_tx)
+    tbox.project.schema_mapping.transformations.set_all(project, all_tx)
+
+    LOGGER.info("Updating the unified dataset...")
+    tbox.project.categorization.jobs.update_unified_dataset(project)
+
+    # Prepare and post labels
+    category_dict = {", ".join(category): category for category in category_list}
+    labels_to_bootstrap = [
+        {
+            "action": "CREATE",
+            "recordId": hash(key),
+            "record": {"verified": {"category": {"path": path}, "reason": "Taxonomy bootstrap"}},
+        }
+        for key, path in category_dict.items()
+    ]
+    project.client.post(
+        f"projects/{project.resource_id}/categorizations/labels:updateRecords",
+        json=labels_to_bootstrap,
+    ).successful()
+    LOGGER.info(f"Created and inserted labels into {project.name}")
```

Comment:
Please create a separate function for this within the toolbox

Reply:
How do you envision this function to look like in terms of the arguments it takes? I can imagine one version to be a dataframe with column names for the path and PK but wanted to check with you to see if you think otherwise.

Reply:
A simple dictionary is probably better in this case

Reply:
As a note, this (`projects/{project.resource_id}/categorizations/labels`) is a dataset, and upserting labels is exactly the same operation as upserting records.  If you can get a `Dataset` instance `ds` of this, this functionality can probably be handled with a `ds.upsert_records()` call.